transport/http.go

```diff
@@ -306,7 +313,10 @@ func (t *HTTPTransport) SetServerURL(u ...*url.URL) error {
 	t.intakeURLs = intakeURLs
 	t.configURLs = configURLs
 	t.profileURLs = profileURLs
-	t.urlIndex = 0
+	atomic.StoreInt32(&t.urlIndex, 0)
```

Comment:
There was an unwritten assumption that SetServerURLs is not called concurrently with other methods: `t.intakeURLs` etc. are set without a lock. Is there some new concurrency that requires the atomic call?

Reply:
Since it wasn't stated anywhere and it seemed like a fairly small change to avoid having a potential data race when the index is increased and `SetServerURL` is called at the same time. 

I can revert if you feel like we don't need to do it, but it seemed like a small price to pay to avoid any potential panics on consumer applications.

Reply:
I'd prefer to be precise when it comes to concurrency: either the method needs to be concurrent-safe, or it doesn't. The atomic implies it does, but the unsynchronised mutation of t.intakeURLs etc. implies it doesn't.

Unless we need it (maybe we do?), I'd prefer to revert.